src/content/plugins.md

```diff
@@ -0,0 +1,325 @@
+    needsCache: false,
+}
+```
+
+`options.name`: Custom name used when debugging/printing stack traces. Broccoli will use the name of your
+plugin if this field is not supplied.
+
+`options.annotation`: In addition to the plugin name, Broccoli uses annotations to provide a descriptive label used
+during debugging/printing stack traces. This is not often set by the plugin author, but by the consumer of the 
+plugin to tell multiple instances of the same plugin apart.
+
+`options.persistentOutput`: If true, the output directory is not automatically emptied between
+(re)builds. By default, a plugin's `outputPath` is emptied before each (re)build ensuring a 
+consistent output state with every build. If `persistentOutput` is set to true, this cleanup
+will not happen and the output directory will persist. This may be useful if your plugin
+implements caching to allow the build method to be skipped. 
+
+`options.needsCache`: Despite the name, `needsCache` doesn't provide caching for the plugin. If true, a directory
+is created for the plugin to store temporary files that may be needed between rebuilds that are not included in the 
+build output. Broccoli sets the path to this directory to `this.cachePath`.  
+
+## Build
+
+The build method is where the grunt (lol, intentional pun) of the work happens. Build can do anything you can do in
+node. Additionally, if `build()` returns a promise, Broccoli will wait until the promise resolves before continuing
+the rest of the build. Broccoli only builds one plugin at a time in order, from top to bottom of the build graph.
+
+This function will typically access the following only properties:
+
+`this.inputPaths`: An array of paths on disk corresponding to each node in inputNodes. Your plugin will read 
+files from these paths.
+
+`this.outputPath`: Broccoli will automatically create a directory for this plugin to write to when it starts up. 
+Your plugin must write files to this path, and Broccoli will use this directory as an `inputPath` to the next 
+plugin. This directory is emptied by Broccoli before each build, unless the `persistentOutput` option is true.
+
+`this.cachePath`: The path on disk to an auxiliary cache directory. Use this to store files that 
+you want preserved between builds but do not end up in your outputPath. This path is only set
+when the `needsCache` option is true, and the directory will only be deleted when the Broccoli 
+process exits.
+
+All paths stay the same between rebuilds.
+
+# Example plugin
+
+Let's build a sample concatenation plugin. It's going to concatenate a set of files matching a glob expression.
+
+```js
+import Plugin from 'broccoli-plugin';
+import walkSync from 'walk-sync';
+import fs from 'fs';
+
+export class ConcatPlugin extends Plugin {
+    constructor(inputNodes, options) {
+        super(inputNodes, options);
+        
+        this.fileMatchers = options.globs || ['**/*'];
+        this.joinSeparator = options.joinSeparator || "\n";
```

Comment:
I can imagine cases where a empty string separator would be desirable here, but since an empty string is falsy, it'd default to a `\n` if you tried and provided one. Not sure if that's worth fixing here if it's a lot of code.